buidl/tx.py

```diff
@@ -409,15 +536,38 @@ def verify_input(self, input_index):
             or script_pubkey.is_p2wsh()
             or (redeem_script and redeem_script.is_p2wsh())
         ):
-            # calculate the z using sig_hash_bip143
-            z = self.sig_hash_bip143(input_index, redeem_script, witness_script)
+
+            def sig_hash(hash_type):
```

Comment:
Is this necessary? Anonymous functions are hard to read/follow, this was a common design pattern of `pybitcointools` back in the day!